aws_topology/stackstate_checks/aws_topology/resources/stepfunction.py

```diff
@@ -0,0 +1,344 @@
+        elif state_type == 'Parallel':
+            self.process_parallel_state(sfn_arn, state_arn, state.get('Branches'))
+        elif state_type == 'Task':
+            self.process_task_state(state_arn, state)
+        self.emit_component(state_arn, 'aws.stepfunction.state', state)
+
+    def process_choice_state(self, sfn_arn, state_arn, choices, default_state):
+        choices = choices or []
+        if default_state:
+            default_state_arn = sfn_arn + ':state/' + default_state
+            self.agent.relation(state_arn, default_state_arn, 'uses service', {})  # can transition to (default)
+        for choice in choices:
+            choice_next = choice.get('Next')
+            if choice_next:
+                choice_arn = sfn_arn + ':state/' + choice_next
+                self.agent.relation(state_arn, choice_arn, 'uses service', {})  # can transition to (choice)
+
+    def process_map_state(self, sfn_arn, state_arn, iterator):
+        iterator = iterator or {}
+        if iterator:
+            start_at = iterator.get('StartAt')
+            states = iterator.get('States') or {}
+            for iterator_state_name, iterator_state in states.items():
+                self.process_state(
+                    sfn_arn,
+                    state_arn,
+                    iterator_state_name,
+                    iterator_state,
+                    iterator_state_name == start_at
+                )
+
+    def process_parallel_state(self, sfn_arn, state_arn, branches):
+        branches = branches or []
+        for branch in branches:
+            start_at = branch.get('StartAt')
+            states = branch.get('States') or {}
+            for branch_state_name, branch_state in states.items():
+                self.process_state(
+                    sfn_arn,
+                    state_arn,
+                    branch_state_name,
+                    branch_state,
+                    branch_state_name == start_at
+                )
+
+    def get_function_reference(self, reference):
+        fn_arn = ''
+        if isinstance(reference, string_types):
+            parts = reference.split(':', -1)
+            if isinstance(parts, list):
+                if len(parts) == 7:
+                    # plain lambda arn
+                    fn_arn = ':'.join(parts)
+                elif len(parts) == 8 and (parts[7].isdigit() or parts[7].lower() == '$latest'):
+                    # TODO lambda versions are not yet supported in StackState so omit version
+                    fn_arn = ':'.join(parts[0:7])
+                elif len(parts) == 8:
+                    # lambda alias arn
+                    fn_arn = ':'.join(parts)
+                elif len(parts) >= 2 and len(parts) <= 4:
+                    # partial lambda arn OR nameonly + version/alias
+                    end = len(parts)
+                    if parts[-1].isdigit() or parts[-1].lower() == '$latest':
+                        end = end - 1
```

Comment:
if you pop this, `parts.pop(-1)` you lose the need for the end variable, so you can just use `len(parts)`